vtex/omnishipping
# CHK-156: Fix validation of field number

#### What is the purpose of this pull request?


This PR fixes the validation of field `number` which was, in some cases, preventing the request to update shipping data. 

This bug was introduced in [this PR](https://github.com/vtex/omnishipping/pull/1335) (:shameonme:), but I can't remember why this was necessary. Anyway, I took the time to test the cases of the aforementioned PR, and it seems fine.

### How should this be manually tested?

- Use this [workspace](https://jeff--vtexgame1geo.myvtex.com/checkout/cart/add?sku=312&qty=1&seller=1&redirect=true&sc=2)
- Proceed to checkout
- Fill profile info
- Use `Av Paulista` as address
- Fill the number field
- Click on "Calcular Entrega"
- You should be presented delivery options

#### Types of changes
